explore/src/main/scala/explore/proposal/ProposalEditor.scala

```diff
@@ -211,6 +210,9 @@ object ProposalEditor {
 
     val abstractView = abstractAligner.view(_.orUnassign)
 
+    val NESIso: Iso[Option[NonEmptyString], String] =
+      Iso[Option[NonEmptyString], String](_.foldMap(_.value))(s => NonEmptyString.from(s).toOption)
```

Comment:
This seems to be a lawful ISO to me, but I haven't tested it. I'm wondering if it might be generally useful, but I'm not sure where it should live, if so.

Reply:
`explore.optics.ModelOptics`?

Or we could generalize it as a `Iso[Option[A Refined P]], A]` for any `P` and put it into `lucuma.core.optics`, where we already have a `refinedPrism`.

Reply:
I don't think it can be generalized that much. With NonEmptyString, the only invalid string is the empty one. With other refinements, there can be more than one A that would be invalid when refined.

Reply:
Yup, you're right, a general `Iso` is not actually possible.